src/main/java/com/amazon/opendistroforelasticsearch/sql/plugin/RestSqlAction.java

```diff
@@ -78,17 +85,19 @@ public String getName() {
     protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) {
         Metrics.getInstance().getNumericalMetric(MetricName.REQ_TOTAL).increment();
         Metrics.getInstance().getNumericalMetric(MetricName.REQ_COUNT_TOTAL).increment();
-        SqlRequest sqlRequest = SqlRequest.NULL;
+
+        LogUtils.addRequestId();
+
         try {
 
             if (!isSQLFeatureEnabled()) {
                 throw new SQLFeatureDisabledException(
-                    "Either opendistro.sql.enabled or rest.action.multi.allow_explicit_index setting is false"
+                        "Either opendistro.sql.enabled or rest.action.multi.allow_explicit_index setting is false"
                 );
             }
 
-            sqlRequest = SqlRequestFactory.getSqlRequest(request);
-            LOG.info("[{}] Incoming request {}: {}", sqlRequest.getId(), request.uri(), sqlRequest.getSql());
+            final SqlRequest sqlRequest = SqlRequestFactory.getSqlRequest(request);
+            LOG.info("[{}] Incoming request {}: {}", LogUtils.getRequestId(), request.uri(), sqlRequest.getSql());
```

Comment:
Though not related to original fix, but can you also remove "Incoming request" wording,  as logically all requests are incoming and it also does not add any value , except for maybe grepping logs.

Reply:
I can remove the word _incoming_, but I think it's there to indicate this it the first log message about this request, so if you for instance scroll through a log file trying to see the end-to-end execution logs for a given request, this will indicate the start. Doesn't really matter much for me, though, I am fine with just keeping the word _request_.

Reply:
Sounds plausible to me. Lets keep it then.